demos/using-dev-build/README.md

```diff
@@ -1,16 +1,63 @@
+- `whoami`: A simple canister that checks that calls are authenticated, and that returns the "principal of the caller". The implementation is terribly simple:
+  ```motoko
+  actor {
+      public query ({caller}) func whoami() : async Principal {
+          return caller;
+      };
+  };
+  ```
+  On the IC, principals are basically the "identity" of someone performing a request or "call" (hence "caller"), and every call must be authenticated — i.e., must have a valid principal. When using Internet Identity you are using [self-authenticating principals](https://smartcontracts.org/docs/interface-spec/index.html#principal), which is a very fancy way of saying that you have a private key on your laptop (hidden behind TouchID, Windows Hello, etc) that your browser uses to sign and prove that you are indeed the person issuing the calls to the IC. If the IC actually lets the call (request) through to the `whoami` canister, it means that everything checked out, and the `whoami` canister just responds with the information the IC adds to requests, namely your identity (principal).
+
+Let's now use those canisters. Don't care about details? Skip to the [helpers](#helpers).
+
+In order to talk to those canisters (for instance to view the webapp in your browser) you need to figure the ID of each canister and then use an URL of the form `https://localhost:8000/?canisterId=<canister ID>` (where `8000` is the port used by `dfx` to proxy calls to the replica; that port is usually specified in the `dfx.json`). You can find the canister IDs in the output of the `dfx command`, or by checking `dfx`'s "internal" (read: non-documented) state:
+
 ```
+~/internet-identity/demos/using-dev-build$ cat .dfx/local/canister_ids.json {
+  "__Candid_UI": {
+    "local": "r7inp-6aaaa-aaaaa-aaabq-cai"
+  },
+  "internet_identity": {
+    "local": "rwlgt-iiaaa-aaaaa-aaaaa-cai"
+  },
+  "webapp": {
+    "local": "rrkah-fqaaa-aaaaa-aaaaq-cai"
+  },
+  "whoami": {
+    "local": "ryjl3-tyaaa-aaaaa-aaaba-cai"
+  }
+```
+
+You might get different canister IDs (and that's totally fine). If the `webapp` canister ID is `rrkah-fqaaa-aaaaa-aaaaq-cai`, you should be able to point your browser to [`http://localhost:8000/?canisterId=rrkah-fqaaa-aaaaa-aaaaq-cai`](http://localhost:8000/?canisterId=rrkah-fqaaa-aaaaa-aaaaq-cai) to see the webapp. Hurray!
+
+![](./webapp.png)
+
+_If you actually use the webapp, make sure that the "Internet Identity URL" field points to `http://localhost:8000/?canisterId=<canister ID of the internet_identity canister>`._
+
+## Helpers
+
+Figuring the canister IDs, and using the `canisterId=...` query parameter is all a bit cumbersome. Here are some commands you might like:
+
+- `npm run start`: Build the app and serve it on `localhost:8080` with hot reload on code changes, ideal for hacking on the webapp.
+- `npm run proxy`: Start a proxy that serves Internet Identity on `localhost:8086` and the webapp on `localhost:8087` for easy access.
```

Comment:
When running it like this, I get redirected to `http://localhost:8000/?canisterId=rwlgt-iiaaa-aaaaa-aaaaa-cai#authorize` for II, not `localhost:8086`.

Reply:
I think that it's easier to leave out this part as implementation of the e2e testing. Rather than adding it here as an explanation.